daisy/target_pool.go

```diff
@@ -0,0 +1,83 @@
+package daisy
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	daisyCompute "github.com/GoogleCloudPlatform/compute-image-tools/daisy/compute"
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+)
+
+var (
+	targetPoolURLRegex = regexp.MustCompile(fmt.Sprintf(`^(projects/(?P<project>%[1]s)/)?regions/(?P<zone>%[2]s)/TargetPools/(?P<targetPool>%[2]s)$`, projectRgxStr, rfc1035))
+)
+
+func (w *Workflow) targetPoolExists(project, zone, targetInstance string) (bool, DError) {
+	return w.targetInstanceCache.resourceExists(func(project, zone string, opts ...daisyCompute.ListCallOption) (interface{}, error) {
+		return w.ComputeClient.ListTargetInstances(project, zone)
+	}, project, zone, targetInstance)
+}
+
+// TargetPool is used to create a GCE targetPool.
+type TargetPool struct {
+	compute.TargetPool
+	Resource
+}
+
+// MarshalJSON is a hacky workaround to compute.TargetPool's implementation.
+func (tp *TargetPool) MarshalJSON() ([]byte, error) {
+	return json.Marshal(*tp)
+}
+
+func (tp *TargetPool) populate(ctx context.Context, s *Step) DError {
+	var errs DError
+	tp.Name, errs = tp.Resource.populateWithGlobal(ctx, s, tp.Name)
+
+	tp.Description = strOr(tp.Description, defaultDescription("TargetPool", s.w.Name, s.w.username))
+	tp.link = fmt.Sprintf("projects/%s/regions/%s/targetPools/%s", tp.Project, getRegionFromZone(s.w.Zone), tp.Name)
+	return errs
+}
+
+func (tp *TargetPool) validate(ctx context.Context, s *Step) DError {
+	pre := fmt.Sprintf("cannot create target-pool %q", tp.daisyName)
```

Comment:
Use a consistent name for target pools (i.e. either target-pool, TargetPool, target pool, but not all of them)

Reply:
Using` target pool` as plain text log and use `TargetPool` as ResourceTypeName when explicit indicate type